### PR TITLE
Instance and storage volume name validation improvements

### PR DESCRIPTION
**Pull request by tomponline**

### CODE REVIEW

File: lxd/instance/instance_utils.go [hunk 1]
```diff
@@ -678,7 +678,7 @@ func ValidName(instanceName string, isSnapshot bool) error {
 
 		// Snapshot part is more flexible, but doesn't allow space or / character.
 		if strings.ContainsAny(snapshotName, " /") {
-			return fmt.Errorf("Invalid instance snapshot name: Cannot contain space or / characters")
+			return fmt.Errorf("Invalid instance snapshot name: Cannot contain spaces or slashes")
```
Comment (roosterfish) [inline lxd/instance/instance_utils.go]: In commit https://github.com/canonical/lxd/pull/13261/commits/1009cf770710c4ba3c9f437d1daaf4388c9e80b3 you are using `%q` to also indicate the name of the snapshot. Would this be applicable here too?

File: lxd/instance/instance_utils.go [hunk 2]
```diff
@@ -682,7 +682,7 @@ func ValidName(instanceName string, isSnapshot bool) error {
 		}
 	} else {
 		if strings.Contains(instanceName, shared.SnapshotDelimiter) {
-			return fmt.Errorf("The character %q is reserved for snapshots", shared.SnapshotDelimiter)
+			return fmt.Errorf("Invalid instance name: Cannot contain slashes")
```
Comment (roosterfish) [inline lxd/instance/instance_utils.go]: To be in line with the other messages, would it make sense to add the name of the instance to the error message too?

**Review by roosterfish — CHANGES_REQUESTED** [review]
Looking good just a small nit to make the error messages consistent.

File: lxd/instance/instance_utils.go (hunk 1 above)
Comment (tomponline) [reply lxd/instance/instance_utils.go]: yeah i think it would

File: lxd/instance/instance_utils.go (hunk 2 above)
Comment (tomponline) [reply lxd/instance/instance_utils.go]: yeah i think it would

Comment (tomponline) [reply lxd/instance/instance_utils.go]: Fixed in https://github.com/canonical/lxd/pull/13261/commits/1009cf770710c4ba3c9f437d1daaf4388c9e80b3

File: lxd/instance/instance_utils.go (hunk 1 above)
Comment (tomponline) [reply lxd/instance/instance_utils.go]: Fixed in https://github.com/canonical/lxd/pull/13261/commits/1009cf770710c4ba3c9f437d1daaf4388c9e80b3

**Comment by roosterfish** [comment]
@tomponline have you pushed the latest changes?

**Comment by tomponline** [comment]
Yes i did before you reviewed.

**Comment by tomponline** [comment]
https://github.com/canonical/lxd/pull/13261/commits/1009cf770710c4ba3c9f437d1daaf4388c9e80b3

File: lxd/instance/instance_utils.go (hunk 1 above)
Comment (tomponline) [reply lxd/instance/instance_utils.go]: Oh sorry i misunderstood your point

Comment (tomponline) [reply lxd/instance/instance_utils.go]: fixed

**Comment by roosterfish** [comment]
Ah sorry, there are two commits, one that changes the message and one that adds the names. I have missed this.

**Review by roosterfish — APPROVED** [review]
LGTM!

**Pull request merged**